docs/usage/string-pattern-matching.md

```diff
@@ -1,49 +1,80 @@
+| `!/^a/`   | matches any string not starting with `a` in lower case                  |
+
+### Use regex101 to test your patterns
 
 If you want to test your patterns interactively online, we recommend [regex101.com](https://regex101.com/?flavor=javascript&flags=ginst).
-Be aware that backslashes (`\`) of the resulting regex have to still be escaped e.g. `\n\s` --> `\\n\\s`. You can use the Code Generator in the sidebar and copy the regex in the generated "Alternative syntax" comment into JSON.
+You can use the Code Generator in the sidebar and copy the regex in the generated "Alternative syntax" comment into JSON.
+
+<!-- prettier-ignore -->
+!!! warning "Escape the backslashes from regex101"
+    Before you copy/paste the regex from regex101 into your Renovate config, you must escape the backslashes (`\`) first.
+    For example: `\n\s` --> `\\n\\s`.
 
 ## Glob matching
 
 If the string provided is not a regex pattern then it will be treated as a glob pattern and parsed using the `minimatch` library.
 Although glob patterns were designed originally for file name matching, many users find glob syntax easier to understand than regex so prefer it.
 
-Glob patterns are evaluated with case _insensitivity_ and this is not configurable, so if you require a case-sensitive pattern then you should use a regex pattern instead.
+### Glob patterns always ignore casing
 
-Examples:
+Glob patterns are always evaluated with case _insensitivity_ and you can not change this.
+If you need a case-sensitive pattern you must use a regex pattern.
 
-- `abc123` matches `abc123` exactly, or `AbC123`.
-- `abc*` matches `abc`, `abc123`, `ABCabc`, etc.
+### Example glob patterns
+
+| Pattern  | Glob pattern explanation                |
+| -------- | --------------------------------------- |
+| `abc123` | matches `abc123` exactly, or `AbC123`   |
+| `abc*`   | matches `abc`, `abc123`, `ABCabc`, etc. |
 
 ## Negative matching
 
 Renovate has a specific approach to negative matching strings.
 
-"Positive" matches are patterns (in glob or regex) which don't start with `!`.
-"Negative" matches are patterns starting with `!` (e.g. `!/^a/` or `!b*`).
+"Positive" matches are patterns (in glob or regex) which do _not_ start with `!`.
+"Negative" matches are patterns starting with `!`, like `!/^a/` or `!b*`.
 
 For an array of patterns to match, the following must be true:
 
-- If any positive matches are included, at least one must match.
-- If any negative matches are included, none must match.
+- If any _positive_ matches are included, at least _one_ must match
+- If any _negative_ matches are included, _none_ must match
+
+For example, the pattern `["/^abc/", "!/^abcd/", "!/abce/"]`:
 
-For example, `["/^abc/", "!/^abcd/", "!/abce/"]` would match "abc" and "abcf" but not "foo", "abcd", "abce", or "abcdef".
+- matches `"abc"` and `"abcf"`
+- does _not_ match `"foo"`, `"abcd"`, `"abce"`, or `"abcdef"`
 
 ## Usage in Renovate configuration options
```

Comment:
I don't know what we really mean with:

> Renovate has matured its approach to string pattern matching over time, but this means that existing configurations may have a mix of approaches and not be entirely consistent with each other.

How can Renovate, the bot, mature? It's just a bunch of code. 🙃 

I think what we really mean is something like: "The maintainers worked on string pattern matching, and have tried different approachers over time. This means that some kinds of string pattern matching are not consistent with other kinds."

In any case, we should be clear:

- who does what to the Renovate string pattern matching
- why it's different
- why the user should care

Also, instead of saying "we have different approaches", could we make things consistent? I assume the maintainers know better what works and doesn't work with the experience over time?

Reply:
The point was to try to explain that differences aren't necessarily because they make sense, but because of legacy reasons. "Evolved" may be a better term than "matured"

Reply:
Evolved seems better. I've rewritten the text.